parsers/ES.py

```diff
@@ -236,30 +238,27 @@ def fetch_production(
 
 @refetch_frequency(timedelta(days=1))
 def fetch_exchange(
-    zone_key1: ZONE_KEYS,
-    zone_key2: ZONE_KEYS,
+    zone_key1: ZoneKey,
+    zone_key2: ZoneKey,
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> List[dict]:
+    sorted_zone_keys = "->".join(sorted([zone_key1, zone_key2]))
```

Comment:
```suggestion
    sorted_zone_keys = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
    check_valid_parameters(sorted_zone_keys, session, target_datetime, logger)
```
To propagate the type.